File: ts_to_heifer/test/heifer_output/03_control_flow.ml
=== Translation Successful ===

Heifer IR:
let abs = fun x (*@ req x:int; ens res:int @*) -> if x<0 then (0 - x) else (()) in
let max = fun a b (*@ req a:int/\b:int; ens res:int @*) -> if a>b then a else (b) in
let min = fun a b (*@ req a:int/\b:int; ens res:int @*) -> if a<b then a else (b) in
let sign = fun x (*@ req x:int; ens res:Str @*) -> if x>0 then "positive" else (if x<0 then "negative" else ("zero")) in
let classify = fun x y (*@ req x:int/\y:int; ens res:Str @*) -> if x>0 then if y>0 then "quadrant_1" else ("quadrant_4") else (if y>0 then "quadrant_2" else ("quadrant_3")) in
let safe_divide = fun a b (*@ req a:int/\b:int; ens res:int @*) -> if b=0 then 0 else (()) in
()

Type: unit

✓ Output is in Heifer's native format

File: ts_to_heifer/test/test_cases/03_control_flow.ts
// Test 3: Control flow statements
// Testing if-else, conditionals, and branching

// Simple if-else
function abs(x: number): number {
  if (x < 0) {
    return -x;
  }
  return x;
}

// If-else with both branches returning
function max(a: number, b: number): number {
  if (a > b) {
    return a;
  } else {
    return b;
  }
}

// Ternary operator
function min(a: number, b: number): number {
  return a < b ? a : b;
}

// Multiple conditions
function sign(x: number): string {
  if (x > 0) {
    return "positive";
  } else if (x < 0) {
    return "negative";
  } else {
    return "zero";
  }
}

// Nested conditionals
function classify(x: number, y: number): string {
  if (x > 0) {
    if (y > 0) {
      return "quadrant_1";
    } else {
      return "quadrant_4";
    }
  } else {
    if (y > 0) {
      return "quadrant_2";
    } else {
      return "quadrant_3";
    }
  }
}

// Early return
function safe_divide(a: number, b: number): number {
  if (b === 0) {
    return 0;
  }
  return a / b;
}
